client/state/current-user/actions.js

```diff
@@ -20,26 +21,44 @@ export function setCurrentUser( user ) {
 	};
 }
 
+let fetchingUser = null;
+
 export function fetchCurrentUser() {
-	return async ( dispatch ) => {
+	return ( dispatch ) => {
+		if ( fetchingUser ) {
+			return fetchingUser;
+		}
+
 		dispatch( {
 			type: CURRENT_USER_FETCH,
 		} );
 
-		try {
-			const user = await wpcom.me().get( {
+		fetchingUser = wpcom
+			.me()
+			.get( {
 				meta: 'flags',
-			} );
-			const userData = filterUserObject( user );
+			} )
+			.then( async ( user ) => {
+				const userData = filterUserObject( user );
 
-			const storedUserId = getStoredUserId();
-			if ( storedUserId != null && storedUserId !== userData.ID ) {
-				await clearStore();
-			}
+				const storedUserId = getStoredUserId();
+				if ( storedUserId != null && storedUserId !== userData.ID ) {
+					await clearStore();
+				}
 
-			setStoredUserId( userData.ID );
-			dispatch( setCurrentUser( userData ) );
-		} catch {}
+				setStoredUserId( userData.ID );
+				dispatch( setCurrentUser( userData ) );
+
+				// @TODO: Remove this once `lib/user` has been fully reduxified
+				userFactory().data = userData;
```

Comment:
You should also `userFactory().emit( 'change' )` to make both methods correspond perfectly.

Reply:
Of course, good call. Done in https://github.com/Automattic/wp-calypso/pull/53814/commits/1e3753073df665f621d84ea4554a07640febb9c3